server/testutil/testhttp/testhttp.go

```diff
@@ -35,3 +35,27 @@ func StartServer(t *testing.T, handler http.Handler) *url.URL {
 	go http.Serve(lis, handler)
 	return u
 }
+
+// TestResponseWriter is an http.ResponseWriter that fails the test if an
+// HTTP error code is written, but otherwise discards all data written to it.
+type TestHttpResponseWriter struct {
```

Comment:
nit: could rename from `testhttp.TestHttpResponseWriter` to `testhttp.ResponseWriter` (and NewTestHttpResponseWriter => NewResponseWriter) to avoid stutter